### PR TITLE
loop 

### PR DESCRIPTION
**Pull request by shiqizng**
This PR contains support for loop

### CODE REVIEW

File: compiler/codegen.go [hunk 1]
```diff
@@ -207,6 +208,31 @@ func (n *ifStatementNode) Codegen(ostream io.Writer) {
 	fmt.Fprintf(ostream, "if_stmt_end_%d:\n", &n)
 }
 
+func (n *forStatementNode) Codegen(ostream io.Writer) {
+	if ids==nil {
```
Comment (algorandskiy) [inline compiler/codegen.go:212]: go fmt please

File: compiler/codegen_test.go [hunk 2]
```diff
@@ -282,12 +283,12 @@ function logic() {
...
-	}
-	a.Empty(errors)
-	prog := Codegen(result)
-	a.Greater(len(prog), 0)
+	//if len(errors) != 0 {
+	//	result.Print()
+	//}
+	//a.Empty(errors)
+	//prog := Codegen(result)
+	//a.Greater(len(prog), 0)
```
Comment (algorandskiy) [inline compiler/codegen_test.go:291]: why disabled?

File: compiler/codegen_test.go [hunk 3]
```diff
@@ -841,3 +842,85 @@ end_main:
...
+		y=y+1
+	}
+	return 1
+}
+`
+	result, errors := Parse(source)
+	a.NotEmpty(result, errors)
+	a.Empty(errors)
+	actual := Codegen(result)
+	fmt.Println(actual)
```
Comment (algorandskiy) [inline compiler/codegen_test.go:900]: not needed

File: TealangParser.g4 [hunk 4]
```diff
@@ -79,7 +81,7 @@ expr
     |   builtinVarExpr                              # BuiltinObject
     |   op=LNOT expr                                # Not
     |   op=BNOT expr                                # BitNot
-    |	expr op=(MUL|DIV|MOD) expr                  # MulDivMod
```
Comment (algorandskiy) [inline TealangParser.g4:82]: tabs/vs spaces?

File: compiler/parser.go [hunk 5]
```diff
@@ -655,23 +677,6 @@ func (l *exprListener) EnterIfExpr(ctx *gen.IfExprContext) {
 	l.expr = listener.getExpr()
 }
 
-func (l *exprListener) EnterCondExpr(ctx *gen.CondExprContext) {
```
Comment (algorandskiy) [inline compiler/parser.go:658]: why did it and how's `IfExprNode` constructed now?

Comment (shiqizng) [reply compiler/parser.go:676]: I got CondIfExpr() for this block of code.  IfExprNode is resolved in EnterIfExpr()

File: compiler/parser.go [hunk 6]
```diff
@@ -996,9 +1003,6 @@ func (l *exprListener) EnterBuiltinVarTupleExpr(ctx *gen.BuiltinVarTupleExprCont
 
 func (l *exprListener) EnterAccountsBalanceExpr(ctx *gen.AccountsBalanceExprContext) {
 	name := "balance"
-	if ctx.MINIMUMBALANCE() != nil {
-		name = "min_balance"
-	}
```
Comment (pzbitskiy) [inline compiler/parser.go:1001]: merge error, this is needed

File: main.go [hunk 7]
```diff
@@ -53,7 +53,7 @@ Syntax highlighter for vscode: https://github.com/pzbitskiy/tealang-syntax-highl
 		if inFile == "-" {
 			data, err := ioutil.ReadAll(os.Stdin)
 			if err != nil {
-				fmt.Println(err.Error())
```
Comment (pzbitskiy) [inline main.go:56]: all changes in this file look as merge errors

File: compiler/codegen_test.go [hunk 8]
```diff
@@ -907,6 +908,85 @@ end_main:
...
+func TestBreak(t *testing.T) {
+	a := require.New(t)
+
+	source := `
+function logic() {
+	let y= 0;
+	for 1 { 
+		if y==10 {break;}
+		y=y+1
+	}
```
Comment (pzbitskiy) [inline compiler/codegen_test.go:956]: please add this into ast_test.go / TestAllFeatures

**Review by pzbitskiy — COMMENTED** [review]
Looks great! Please fix merge errors and run `make test` and ensure it runs well.

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**